numbat/examples/inspect.rs

```diff
@@ -86,15 +79,126 @@ fn inspect_functions_in_module(ctx: &Context, module: String) {
         println!("{signature}");
         println!("```");
         println!();
+
+        if !examples.is_empty() {
+            println!("<details>");
+            println!("<summary>Examples</summary>");
+            println!();
+
+            for (example_code, example_description) in examples {
+                let mut example_ctx = prepare_context();
+                let _result = example_ctx
+                    .interpret("use prelude", CodeSource::Internal)
+                    .unwrap();
```

Comment:
This could be done just once for the entire program. Then we could create a new example context by cloning an existing one (where the prelude has already been included). Parsing, type checking, interpreting the prelude for every single example seems a bit wasteful and currently takes ~50 ms, which would lead to an overhead of 5 seconds for 100 examples.

Reply:
Yes, that makes sense.